interactive/shell.py

```diff
@@ -145,3 +146,39 @@ def show(self, test_set):
             self.tt._tprint(items)
         else:
             self.err()
+
+    @line_magic
+    def cache(self, line, ident='pytest/interactive'):
+        """
+            self.err("You must provide a key:"
+                     " `cache <local name> as <key>`")
+        """
+        cachedict = self.tt.get_cache_dict()
+        if line:
+            name, _, cache_key = line.partition(' as ')
+
+            # create a new entry
+            if cache_key:
+                testset = self.ns_eval(name)
+                self.tt.set_cache_items(cache_key, testset)
+
+                # update the local shell's ns
+                if testset:
+                    self.shell.user_ns[cache_key] = testset
+                return
+
+            # delete an entry
+            name, delim, cache_key = line.partition('del ')
+            if delim:
+                self.tt.set_cache_items(cache_key, None)  # delete
+                self.shell.user_ns.pop(cache_key)
+                self.tr.write(
+                    "Deleted cache entry for '{}'\n".format(cache_key))
+                return
```

Comment:
Take a step back and spec out how you want to parse the command:

```
operation = "add" | "del"
identifier = [A-Za-z0-9]
cmd = "cache" operation identifier ["as" identifier]
```

something like that, and unambigious. I'm not sure if you require "as", but if you don't, for example, what is the actual error here: `'cache del as'`? Incorrect name? or missing identifier next to the as?